dnsimple/response.py

```diff
@@ -65,6 +65,8 @@ def add_data(self, obj, http_response):
             self.__class__.data = None
         elif 'data' not in raw_json:
             self.__class__.data = obj(raw_json)
+        elif type(raw_json.get('headers')) is list:
+            self.__class__.data = return_list_with_headers_of(obj, raw_json.get('data'), raw_json.get('headers'))
```

Comment:
Seeing this makes me wonder if we considered having the schema as something similar to:

```json
{
  "data": {
     "rows": [],
     "headers": [],
     ...
  },
  "pagination": {...}
}
```

This way we can avoid extending each client's response handling logic for a single endpoint and continue with the response struct approach.

Reply:
It sounds reasonable. Good idea. It will make it even more clear that headers are not request headers but part of the endpoint response payload.